flink-runtime/src/main/java/org/apache/flink/runtime/metrics/util/MetricUtils.java

```diff
@@ -366,6 +369,17 @@ private static void instantiateCPUMetrics(MetricGroup metrics) {
         }
     }
 
+    public static String truncateOperatorName(String operatorName, int maxLength) {
```

Comment:
could we have a unit test for this method please.

Reply:
if maxlength is less than the Suffix length and there is a suffix in the data, shouldn't this be an error as the 2 values cannot be reconciled?

Reply:
> could we have a unit test for this method please.

good catch